src/main/scala/com/scalapenos/riak/internal/RiakHttpClientHelper.scala

```diff
@@ -68,14 +67,15 @@ private[riak] class RiakHttpClientHelper(system: ActorSystem) extends RiakUriSup
     }
   }
 
-  def fetch(server: RiakServerInfo, bucket: String, key: String, resolver: RiakConflictsResolver): Future[Option[RiakValue]] = {
-    httpRequest(Get(KeyUri(server, bucket, key))).flatMap { response ⇒
+  def fetch(server: RiakServerInfo, bucket: String, key: String, resolver: RiakConflictsResolver, conditionalParams: Seq[ConditionalRequestParam] = Seq()): Future[Option[RiakValue]] = {
+    httpRequest(Get(KeyUri(server, bucket, key)).withHeaders(conditionalParams.map(_.asHttpHeader): _*)).flatMap { response ⇒
       response.status match {
-        case OK              ⇒ successful(toRiakValue(response))
-        case NotFound        ⇒ successful(None)
-        case MultipleChoices ⇒ resolveConflict(server, bucket, key, response, resolver).map(Some(_))
-        case other           ⇒ throw new BucketOperationFailed(s"Fetch for key '$key' in bucket '$bucket' produced an unexpected response code '$other'.")
-        // TODO: case NotModified => successful(None)
+        case OK                 ⇒ successful(toRiakValue(response))
+        case NotFound           ⇒ successful(None)
+        case NotModified        ⇒ successful(None) // This means that client is not able to distinguish cases when value is not in Riak or a supplied condition is not met.
```

Comment:
Could you add a way to distinguish between the not-found and not-modified case? Perhaps in a different method to retain backwards compatibility.